### PR TITLE
Add docstring to `on_event`

### LINKED ISSUE #10671: Unclear which Events `on_event` is compatible with
When I first tried using `on_event`, my expectation was that all Events should be accessible from `on_event`, or I should at least get a warning.  However, this is not the case; see discussion with [Bryan](https://discourse.bokeh.org/t/trouble-with-on-event-for-renderers/6711).  

As a quick fix, I think it would be nice if the docstring for `on_event` explicitly laid out this limitation, as while the existing [docs](https://docs.bokeh.org/en/latest/docs/reference/events.html) do imply it, it wasn't at all obvious to me.

### PR DESCRIPTION
**Pull request by nitrocalcite**
Proposal to fix #10671

### CODE REVIEW

File: bokeh/util/callback_manager.py [hunk 1]
```diff
@@ -49,6 +49,11 @@ def __init__(self, *args, **kw):
         self._event_callbacks = dict()
 
     def on_event(self, event, *callbacks):
+        ''' Run callbacks when the specified event occurs on this Model
+
+        Not all ``Event``s are supported for all ``Model``s.
+        See specific ``Event``s in ``bokeh.events`` for more information on which ``Model``s can trigger them.
```
Comment (bryevdv) [inline bokeh/util/callback_manager.py:55]: Sphinx is getting confused, and also it's harder to read as text, so I'd suggest removing all the backticks above. A nice touch would also make ``bokeh.events`` be a link using a Sphinx reference 
```
:ref:`bokeh.events`
```
Lastly although it's not 100% enforced we do try to keep docstrings in particular to about 80 characters wide.

**Comment by bryevdv** [comment]
@nitrocalcite thanks for the PR and changes, one last edit needed, our linter disallows trailing whitepace
```
E         File contains trailing whitespace: bokeh/util/callback_manager.py, line 55.

```

**Review by bryevdv — APPROVED** [review]

**Comment by nitrocalcite** [comment]
No problem, thanks a lot for walking me through the process!

**Comment by bryevdv** [comment]
unit test failure is spurious and unrelated, I will go ahead and merge, thanks @nitrocalcite !

**Pull request merged**